src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            arrayIncrease();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (array.length < size + 1) {
+            arrayIncrease();
+        }
+        System.arraycopy(array, index, array,
+                index + 1, size - index);
+        array[index] = value;
+        size++;
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException(some message here);
        }
        System.arraycopy(array, index, array,
                index + 1, size - index);
        array[index] = value;
        size++;
```
це ще можна спростити так як ми використовуємо arraycopy